protocols/kad/src/query.rs

```diff
@@ -308,13 +341,40 @@ impl<TInner> Query<TInner> {
         state
     }
 
+    /// Tries to (gracefully) finish the query prematurely, providing the peers
+    /// that are no longer of interest for further progress of the query.
+    ///
+    /// A query may require that in order to finish gracefully a certain subset
+    /// of peers must be contacted. E.g. in the case of disjoint query paths a
+    /// query may only finish gracefully if every path contacted a peer whose
+    /// response permits termination of the query. The given peers are those for
+    /// which this is considered to be the case, i.e. for which a termination
+    /// condition is satisfied.
+    ///
```

Comment:
```suggestion
    ///
    /// Returns `true` if the query did indeed finish, `false` otherwise. In the latter case,
    /// a new attempt at finishing the query may be made with new `peers`.
    ///
```